### ISSUE TITLE
New pip version breaking

### ISSUE THREAD

**Issue by nilsleh** [aleximmer/Laplace#258]
Hi, we noticed that the new version 0.2.2 is breaking things in our [unit tests](https://github.com/lightning-uq-box/lightning-uq-box/pull/259) for a domain library where we use this nice laplace library. 

I also tried the following in a separate clean environment:

```shell
pip install -U laplace-torch
```

It appears that with the upgraded version, the laplace directory is not included in `site-packages`, only `laplace_torch-0.2.2.dist-info` (using a conda environment with pip install).

and get the same error:

```
python 3.10.14 (main, May  6 2024, 19:42:50) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import laplace
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'laplace'
```
```shell
pip install laplace-torch==0.2.1
```
Does not yield the error.

**Comment by aleximmer** [comment]
Sorry about this. @wiseodd do you know what's missing or what could have caused this? I couldn't find any problem in the recent changes. Maybe something went wrong with the deployment using uv?

**Comment by wiseodd** [comment]
Fixed here a4fd3e5759367672cbc67a1c58acf5166cb31b16

It seems the `pdm` backend is more reliable. Maybe this issue is a good kick for us to move to a more modern toolchain.

@nilsleh can you test again? `laplace-torch` should be version `0.2.2.2`. If it works, feel free to close the issue!

**Comment by nilsleh** [comment]
@wiseodd Yes, unit tests are passing again. Thanks for the quick response :)
